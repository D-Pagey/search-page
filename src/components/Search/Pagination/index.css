.pagination-container {
  align-items: center;
  flex-direction: column;
  display: flex;
  margin-bottom: 1rem;
  width: 290px;
}

.pagination-results {
  margin: 0;
}

.pagination-btn-container {
  height: 40px;
  margin-top: 10px;
  width: 200px;
}

.page-list {
  display: flex;
  height: inherit;
  justify-content: space-evenly;
  list-style: none;
  margin: 0;
  padding: 0;
}

.page-item {
  display: flex;
}

.page-btn {
  background-color: white;
  border: 0;
  border-radius: 6px;
  padding: 10px;
}

.active {
  background-color: #50ff7d;
}