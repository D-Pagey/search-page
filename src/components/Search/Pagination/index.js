import React from 'react';
import { func, objectOf, any } from 'prop-types';

import './index.css';

export default function Pagination({ handleChange, userParams, totalCount }) {
  const { page } = userParams;

  return (
    <div className="pagination-container">
      <p className="pagination-results">
        {`Showing ${page}-${page}0 of
        ${totalCount ? totalCount.total_count : 0} results`}
      </p>
      <div className="pagination-btn-container">
        <ul className="page-list">
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 1 ? 'active' : null}`}
              name="page"
              value="1"
              onClick={handleChange}
            >
            1
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 2 ? 'active' : null}`}
              name="page"
              value="2"
              onClick={handleChange}
            >
            2
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 3 ? 'active' : null}`}
              name="page"
              value="3"
              onClick={handleChange}
            >
            3
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 4 ? 'active' : null}`}
              name="page"
              value="4"
              onClick={handleChange}
            >
            4
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 5 ? 'active' : null}`}
              name="page"
              value="5"
              onClick={handleChange}
            >
            5
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
}

Pagination.propTypes = {
  handleChange: func.isRequired,
  userParams: objectOf(any).isRequired,
  totalCount: objectOf(any),
};

Pagination.defaultProps = {
  totalCount: {},
};

/** To Do
 * Create page buttons based off of total results
 * Trigger rerender the page based off page number
 */
