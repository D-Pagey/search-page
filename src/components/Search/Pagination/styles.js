import styled from 'styled-components';

export const PaginationWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-bottom: 70px;
  width: 100%;
  @media (min-width: 425px) {
    margin-bottom: 25px;
  }
`;

export const PaginationResults = styled.p`
  margin: 0;
`;

export const PaginationBtnWrapper = styled.div`
  margin-top: 10px;
  width: 232px;
  @media (min-width: 425px) {
    width: 323px;
  }
  @media (min-width: 768px) {
    width: 460px;
  }
`;

export const PageList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  height: inherit;
  justify-content: flex-start;
  list-style: none;
  margin: 0;
  padding: 0;
`;

export const PageItem = styled.li`
  display: flex;
  margin: 3px;
`;

/* eslint-disable */
export const PageButton = styled.button`
  background-color: ${props => props.active ? '#50ff7d' : 'white'};
  border: 0;
  border-radius: 6px;
  padding: 12.5px 5px;
  width: 2.5rem;
`;
/* eslint-enable */
