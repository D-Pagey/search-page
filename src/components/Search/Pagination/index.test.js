import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '.';

it('should render', () => {
  const wrapper = shallow(<Pagination />);
  expect(wrapper).toMatchSnapshot();
});
