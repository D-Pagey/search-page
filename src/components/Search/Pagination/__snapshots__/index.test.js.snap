// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="pagination-container"
>
  <p
    className="pagination-results"
  >
    Showing NaN-NaN of
            undefined results
  </p>
  <div
    className="pagination-btn-container"
  >
    <ul
      className="page-list"
    />
  </div>
</div>
`;
